subgraphs/ribbon-finance/src/common/utils.ts

```diff
@@ -0,0 +1,162 @@
+import { RibbonThetaVaultWithSwap as VaultContract } from "../../generated/ETHCallV2/RibbonThetaVaultWithSwap";
+
+export function equalsIgnoreCase(a: string, b: string): boolean {
+  return a.replace("-", "_").toLowerCase() == b.replace("-", "_").toLowerCase();
+}
+
+export function enumToPrefix(snake: string): string {
+  return snake.toLowerCase().replace("_", "-") + "-";
+}
+
+export function readValue<T>(
+  callResult: ethereum.CallResult<T>,
+  defaultValue: T
+): T {
+  return callResult.reverted ? defaultValue : callResult.value;
+}
+
+export function getVaultBalance(
+  vaultAddress: Address,
+  decimals: number
+): BigDecimal {
+  const vaultContract = VaultContract.bind(vaultAddress);
+
+  const vaultBalance = readValue<BigInt>(
+    vaultContract.try_totalBalance(),
+    constants.BIGINT_ZERO
+  ).divDecimal(constants.BIGINT_TEN.pow(decimals as u8).toBigDecimal());
+  return vaultBalance;
+}
+
+export function updateProtocolAfterNewVault(vaultAddress: Address): void {
+  const protocol = getOrCreateYieldAggregator();
+  const vaultIds = protocol._vaultIds;
+  vaultIds!.push(vaultAddress.toHexString());
+  protocol._vaultIds = vaultIds;
+  protocol.totalPoolCount += 1;
+  protocol.save();
+}
+
+export function updateProtocolTotalValueLockedUSD(): void {
+  const protocol = getOrCreateYieldAggregator();
+
+  const vaultIds = protocol._vaultIds!;
+  let totalValueLockedUSD = constants.BIGDECIMAL_ZERO;
+
+  for (let vaultIdx = 0; vaultIdx < vaultIds.length; vaultIdx++) {
+    const vault = Vault.load(vaultIds[vaultIdx]);
+
+    if (!vault) continue;
+
+    totalValueLockedUSD = totalValueLockedUSD.plus(vault.totalValueLockedUSD);
+  }
+  protocol.totalValueLockedUSD = totalValueLockedUSD;
+  protocol.save();
+}
+
+export function getTokenDecimals(tokenAddr: Address): BigDecimal {
+  const token = ERC20Contract.bind(tokenAddr);
+
+  const decimals = readValue<number>(
+    token.try_decimals(),
+    constants.DEFAULT_DECIMALS.toI64()
+  );
+
+  return constants.BIGINT_TEN.pow(decimals as u8).toBigDecimal();
+}
+
+export function getVaultPricePerShare(vaultAddress: Address): BigDecimal {
+  const vaultContract = VaultContract.bind(vaultAddress);
+  const vaultDecimals = readValue(vaultContract.try_decimals(), 18);
+
+  const vaultPricePerShare = readValue(
+    vaultContract.try_pricePerShare(),
+    constants.BIGINT_ZERO
+  ).divDecimal(constants.BIGINT_TEN.pow(vaultDecimals as u8).toBigDecimal());
+
+  if (vaultPricePerShare.notEqual(constants.BIGDECIMAL_ZERO))
+    return vaultPricePerShare;
+
+  const totalTokensDeposits = readValue(
+    vaultContract.try_totalBalance(),
+    constants.BIGINT_ZERO
+  ).divDecimal(constants.BIGINT_TEN.pow(vaultDecimals as u8).toBigDecimal());
+
+  const totalOptions = readValue(
+    vaultContract.try_totalSupply(),
+    constants.BIGINT_ZERO
+  ).divDecimal(constants.BIGINT_TEN.pow(vaultDecimals as u8).toBigDecimal());
+
+  const pricePerShare = totalTokensDeposits.div(totalOptions);
```

Comment:
I think we can remove this code now, as we are removing the V1 vaults because this logic was explicitly for the V1 vaults as they don't have a method to get the pricePerShare directly.